plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -364,10 +365,29 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
         catch (AmazonServiceException e) {
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
-        dropTableData(table.io(), table.operations().current());
+        try {
+            dropTableData(table.io(), table.operations().current());
```

Comment:
I am expecting an exception not from `table.operations.current()` but from `dropTableData` method. 
`dropTableData` basically gets the snapshots from current metadata and iterates over it. with each snapshot files it loads all manifest files associated with it. If snapshot file is missing then it will throw `org.apache.iceberg.exceptions.NotFoundException` exception.

use case: When a snapshot file is missing then `dropTableData` will throw `org.apache.iceberg.exceptions.NotFoundException`. [`testDropTableWithMissingSnapshotFile`](https://github.com/trinodb/trino/pull/16674/files#diff-3f26b931a71ef29adbdba7ea025c513ecb76eeaa4578ad45cb3f05f254405762R520) covers this case.

Reply:
Thank you for pointing this out.